Releasing.md

```diff
@@ -0,0 +1,48 @@
+# Releasing
+
+This is notes for the current process of releasing a new PyO3 version. Replace `<version>` in all instructions below with the new version.
+
+## 1. Prepare the release commit
+
+1. Replace all instances of the PyO3 current version with the new version to be released. Places to check:
+   - `Cargo.toml` for all PyO3 crates in the repository.
+   - PyO3 version embedded into documentation like the README.
+   - `pre-script.rhai` templates for the examples.
+   - `[towncrier]` section in `pyproject.toml`.
+
+   Places where you need to be a bit careful:
+   - Add _new_ lines to the netlify `_redirects` file rather than replacing the existing ones.
+   - Make sure not to modify the CHANGELOG during this step!
+
+2. Run `towncrier build` to generate the CHANGELOG. The version used by `towncrier` should automatically be correct because of the update to `pyproject.toml` in step 1.
+
+3. Manually edit the CHANGELOG for final notes. Steps to do:
+   - Adjust wording of any release lines to make them clearer for users / fix typos.
+   - Add a new link at the bottom for the new version, and update the `Unreleased` link.
+
+4. Create the commit containing all the above changes, with a message of `release: <version>`. Push to `release-<version>` branch on the main PyO3 repository.
+
+## 2. Create the release PR and draft release notes
+
+Open a PR for the branch, and confirm that it passes CI.
+
+On https://github.com/PyO3/pyo3/releases, click "Draft a new release". The tag will be a new tag of `v<version>` (note preceding `v`) and target should be the `release-<version>` branch you just pushed.
+
+Write release notes which match the style of previous releases. You can get the list of contributors by running `nox -s contributors -- v<prev-version> release-<version>` to get contributors from the previous version tag through to the branch tip you just pushed.
```

Comment:
Maybe it would be good to slightly reword this to indicate that the Nox session will only _after_ pushing to GitHub as it uses the GitHub API instead of the local repository state?

Reply:
Good idea, done 👍